File: 프로그래머스/lv2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    hashmap = {}
    for x in phone_book:
        hashmap[x] = 1
    
    for x in phone_book:
        tmp = ""
        for i in x:
            tmp += i
            if tmp == x:
                break
            if tmp in hashmap and hashmap[tmp] == 1:
                return False
            
    return True

File: 프로그래머스/lv2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        HashMap<String, Integer> map = new HashMap<String, Integer>();
        for(String number : phone_book)
            map.put(number, 1);
        
        for(String number : phone_book){
                for(int i=0; i<number.length(); i++){ // number.length()를 포함하게 되면 모두 false를 반환
                    if(map.containsKey(number.substring(0, i))){
                        return false;
                    }
                }
        }
        
        return true;
    }
}